File: Chapter3.CoffeeShop.Java/src/Condiments/SteamedMilk.java
package Condiments;

import Beverages.Beverage;

public class SteamedMilk extends CondimentDecorator{
    public SteamedMilk(Beverage beverage) {
        this.beverage = beverage;
    }

    @Override
    public double cost() {
        Double pricing = switch (beverage.getSize()) {
            case TALL -> 0.05;
            case GRANDE -> 0.10;
            case VENTI -> 0.15;
        };
        return beverage.cost() + pricing;
    }
    @Override
    public String getDescription() {
        return beverage.getDescription() + ", Steamed Milk";
    }
}

File: Chapter3.CoffeeShop/Models/Condiments/SteamedMilk.cs
namespace Chapter3.CoffeeShop.Models;

public class SteamedMilk : CondimentDecorator
{
    public SteamedMilk(Beverage beverage)
    {
        _beverage = beverage;
    }
    
    public override double Cost()
    {
        var price = 0.10;

        var cost = _beverage.GetSize() switch
        {
            Size.Tall => price - (price / 3),
            Size.Grande => price,
            Size.Venti => price + (price / 3),
            _ => throw new ArgumentOutOfRangeException()
        };

        return _beverage.Cost() + cost;
    }
    public override string GetDescription() => _beverage.GetDescription() + ", Steamed Milk";
}